src/libraries/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -200,10 +195,14 @@ public static string IdnEquivalent(string hostname)
         {
             // check if only ascii chars
             // special case since idnmapping will not lowercase if only ascii present
-            if (IsAscii(hostname))
+            if (Ascii.IsValid(hostname))
             {
                 // just lowercase for ascii
-                return hostname.ToLowerInvariant();
+                return string.Create(hostname.Length, hostname, static (chars, asciiHostName) =>
```

Comment:
Why is this code better than what was there before?  We still want to avoid the allocation in the case where the string is already all lowercased.